File: sample/venomtyme.kt
/*
 * Once upon a time ...
 */

class Cat(
  val location: String,
  val birthDate: Int,
  val currentDate: Int,
  val catlikes: Array<String>
) {
  val age: Int
    get() = this.calcAge()

  fun calcAge() =
    this.currentDate - this.birthDate
}

// ... there was a cat named Katlyn

fun main() {
  Cat(
    "Cataliputra",
    2016,
    2022,
    arrayOf("Tasty Food", "Long Naps")
  )
}

File: sample/venomtyme.rb
#
# Once upon a time...
#

class Cat
  def initialize(opts)
    @location = opts[:location]
    @birthDate = opts[:birthDate]
    @currentDate = opts[:currentDate]
    @catlikes = opts[:catlikes]
  end

  def age
    calcAge
  end

  private

  def calcAge
    @currentDate - @birthDate
  end
end

# ...there was a cat named Katlyn

catppuccin = Cat.new(
  location: 'Cataliputra',
  birthDate: 2016,
  currentDate: 2022,
  catlikes: %w[Tasty Food Long Naps]
)

puts catppuccin.age
